graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/fetchers/deployed/MutationFetcher.java

```diff
@@ -32,6 +49,162 @@ protected MutationFetcher(MutationModelT model, MappingModel mappingModel) {
     this.model = model;
   }
 
+  protected abstract MutationPayload<ResultT> getPayload(
+      DataFetchingEnvironment environment, StargateGraphqlContext context)
+      throws UnauthorizedException;
+
+  @Override
+  protected CompletableFuture<ResultT> get(
+      DataFetchingEnvironment environment, StargateGraphqlContext context) {
+
+    MutationPayload<ResultT> payload = null;
+    Parameters parameters = null;
+    Exception buildException = null;
+    try {
+      payload = getPayload(environment, context);
+      parameters = buildParameters();
+    } catch (Exception e) {
+      buildException = e;
+    }
+
+    if (isAtomic(environment, payload)) {
+      return executeAsPartOfBatch(payload, parameters, buildException, environment, context);
+    } else {
+      return executeAsIndividualQueries(payload, parameters, buildException, context);
+    }
+  }
+
+  /**
+   * Processes the current operation if it is part of a larger atomic mutation, that requires
+   * executing all the queries in a single CQL batch. For example:
+   *
+   * <pre>
+   * mutation @atomic {
+   *   operation1(args) { ... }
+   *   operation2(args) { ... }
+   * }
+   * </pre>
+   *
+   * Our fetchers only cover a single operation, so we need to accumulate queries before we can
+   * execute the batch.
+   */
+  private CompletableFuture<ResultT> executeAsPartOfBatch(
+      MutationPayload<ResultT> payload,
+      Parameters parameters,
+      Exception buildException,
+      DataFetchingEnvironment environment,
+      StargateGraphqlContext context) {
+    StargateGraphqlContext.BatchContext batchContext = context.getBatchContext();
+    if (buildException != null) {
+      batchContext.setExecutionResult(buildException);
+    } else if (!batchContext.setParameters(parameters)) {
+      batchContext.setExecutionResult(
+          new GraphQLException(
```

Comment:
Is consistency level the only parameter that could be different?

the `cqlfirst` implementation uses the following error. Is this different?

````
  buildException =
          new GraphQLException("options can only de defined once in an @atomic mutation selection")

```

Reply:
Yes, the CLs are the only ones that can change, and there is no `options` objects in this version of the API so I changed the message so that it would be less confusing.

The only other mutation option in GraphQL-first is TTL, and it's appended directly in the CQL query string, so it can actually be different across queries in the same batch.